### PR TITLE
take into account all promoted groups with search relevance bumps

### LINKED ISSUE #15274: add promoted sorting to addons search api
We can currently sort by recommended status (and it's the default).  We want to expand that to be promoted, which will be largely the same, but also bump the search ranking of some non-recommend promotional add-ons somewhat (https://docs.google.com/spreadsheets/d/1ENbv1CuLDd25Sk0XAwgbWTl9is5rV12OxJD6g1kUi48/edit#gid=1788006942 specifies the Verified Tier 1 group - aka Sponsored - to have a search boost)

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #15274 - certainly one for @diox to look over when he's back.

The main change here is in filter.py.  What's happening with the changes in test_filters.py is whenever you add something to the ES index (I added 3 extra addons in the setup), or change any of the items, the scores slightly change.  It's regarded as "reasonable" if the results are in the same order and the new score is not too different than the old score.

### CODE REVIEW

**Comment by eviljeff** [comment]
... and the tests giving different scores on travis than locally 🙄 - we've had this before but I can't remember what we did to fix it.

**Comment by eviljeff** [comment]
Note: Product had a last-minute change of mind so I need to rework this (but I'd ideally still like the code to support different bumps for different groups to land)

**Review by diox — COMMENTED** [review]
As discussed on Matrix the search ranking tests failures are likely timing issues caused by indexing happening too late. Making the add-on promoted before enabling the signals back would likely solve it

File: src/olympia/search/filters.py [hunk 1]
```diff
@@ -754,13 +755,24 @@ def apply_search_query(self, search_query, qs, sort=None):
...
+            reverse=True)
+        for bump, promo_ids in ranking_bump_groups:
+            if not bump:
+                continue
+            functions.append(
+                query.SF({
+                    'weight': bump,
+                    'filter': (
+                        Q('terms', **{
+                            'promoted.group_id': [p.id for p in promo_ids]})
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/search/filters.py:772]: @bobsilverberg landed the changes that added the properties to the index - I _think_ there was a reindex?

**Comment by eviljeff** [comment]
Product decided against search boosts for now so I removed those constants (and the test changes that support them).

File: src/olympia/search/filters.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/search/filters.py:772]: We triggered a reindex on 2020-09-01.

**Review by diox — APPROVED** [review]

**Pull request merged**